lib/cocoapods/sandbox/path_list.rb

```diff
@@ -52,9 +52,18 @@ def read_file_system
         end
         escaped_root = escape_path_for_glob(root)
 
-        absolute_paths = Pathname.glob(escaped_root + '**/*', File::FNM_DOTMATCH).lazy
-        dirs_and_files = absolute_paths.reject { |path| path.basename.to_s =~ /^\.\.?$/ }
-        dirs, files = dirs_and_files.partition { |path| File.directory?(path) }
+        absolute_paths = Pathname.glob(escaped_root + '**/*', File::FNM_DOTMATCH)
+        absolute_paths = absolute_paths.reject { |path| path.basename.to_s =~ /^\.\.?$/ }.sort_by(&:to_s)
+        symbolic_paths, absolute_paths = absolute_paths.partition { |path| File.symlink?(path) }
+        expanded_symbolic = symbolic_paths.map do |path|
+          begin
+            path.realpath
+          rescue Errno::ENOENT
+            raise Informative, "Unable to find reference for symbolic link #{path} pointing to #{File.readlink(path)}"
```

Comment:
Unfortunately the prior error message from the project was able to include the target name, but here there's no way to access that. I added the destination of the symlink hoping it would alleviate that, but open to suggestions here